diff --git a/Doc/conf.py b/Doc/conf.py
index 4cca13b..87f1e22 100644
--- a/Doc/conf.py
+++ b/Doc/conf.py
@@ -73,6 +73,11 @@ html_theme_options = {
     'root_include_title': False   # We use the version switcher instead.
 }
 
+# Override stylesheet fingerprinting for Windows CHM htmlhelp to fix GH-91207
+# https://github.com/python/cpython/issues/91207
+if any('htmlhelp' in arg for arg in sys.argv):
+    html_style = 'pydoctheme.css'
+
 # Short title used e.g. for <title> HTML tags.
 html_short_title = '%s Documentation' % release
 
